app/controllers/projects/archive_controller.rb

```diff
@@ -53,12 +54,18 @@ def change_status_action(status)
   end
 
   def change_status(status)
-    "Projects::#{status.to_s.camelcase}Service"
-      .constantize
+    service_class(status)
       .new(user: current_user, model: @project)
       .call
   end
 
+  def service_class(status)
+    case status
+    when :archive then Projects::ArchiveService
+    when :unarchive then Projects::UnarchiveService
+    end
+  end
+
   def project_path_with_status
     acceptable_params = params.permit(:status).to_h.compact.select { |_, v| v.present? }
```

Comment:
@ulferts What is this `status` parameter? Is it really used?

Reply:
Good point. It looks to be a relic from when what is now the boolean property `active` was still the enum (with only two values) property `status`. My first urge would then be to simply change it to `active` but on the other hand, the code seems to work without that value, so it is worth to check whether `permit(:status)` can simply be removed.

Reply:
@ulferts It looks like it was used to filter the projects list to get active / archived projects ([39f53d2a00](https://github.com/opf/openproject/commit/39f53d2a00#diff-55c87ec2e453f92d3fe7bf0ec3369e8488b2b24621d8259e079ddc3875903c0aR305-R306)). And then on [4e5b9afef5](url), the status enum field was replaced by the active boolean field, and `params[:filters]` [started to be used](https://github.com/opf/openproject/commit/4e5b9afef5#diff-55c87ec2e453f92d3fe7bf0ec3369e8488b2b24621d8259e079ddc3875903c0aR249).

If I'm correct, we can probably remove this `project_path_with_status` and replace it with `projects_path`, right?

Reply:
I'll refactor this once merged.